rviz_default_plugins/src/rviz_default_plugins/displays/depth_cloud/depth_cloud_display.cpp

```diff
@@ -0,0 +1,602 @@
+
+  // Queue size property
+  queue_size_property_ =
+    new rviz_common::properties::IntProperty(
+    "Queue Size", queue_size_,
+    "Advanced: set the size of the incoming message queue.  Increasing this "
+    "is useful if your incoming TF data is delayed significantly from your"
+    " image data, but it can greatly increase memory usage if the messages "
+    "are big.",
+    this, SLOT(updateQueueSize()));
+  queue_size_property_->setMin(1);
+
+  use_auto_size_property_ = new rviz_common::properties::BoolProperty(
+    "Auto Size", true,
+    "Automatically scale each point based on its depth value and the camera parameters.", this,
+    SLOT(updateUseAutoSize()));
+
+  auto_size_factor_property_ =
+    new rviz_common::properties::FloatProperty(
+    "Auto Size Factor", 1, "Scaling factor to be applied to the auto size.",
+    use_auto_size_property_, SLOT(updateAutoSizeFactor()), this);
+  auto_size_factor_property_->setMin(0.0001f);
+
+  use_occlusion_compensation_property_ =
+    new rviz_common::properties::BoolProperty(
+    "Occlusion Compensation", false,
+    "Keep points alive after they have been occluded by a closer point. "
+    "Points are  removed after a timeout or when the camera frame moves.",
+    this, SLOT(updateUseOcclusionCompensation()));
+
+  occlusion_shadow_timeout_property_ = new rviz_common::properties::FloatProperty(
+    "Occlusion Time-Out", 30.0f,
+    "Amount of seconds before removing occluded points from the depth cloud",
+    use_occlusion_compensation_property_, SLOT(updateOcclusionTimeOut()), this);
+}
+
+void DepthCloudDisplay::onInitialize()
+{
+  auto rviz_ros_node_ = context_->getRosNodeAbstraction();
+
+  depthmap_it_.reset(new image_transport::ImageTransport(rviz_ros_node_.lock()->get_raw_node()));
+  rgb_it_.reset(new image_transport::ImageTransport(rviz_ros_node_.lock()->get_raw_node()));
+
+  // Instantiate PointCloudCommon class for displaying point clouds
+  pointcloud_common_ = new PointCloudCommon(this);
+
+  updateUseAutoSize();
+  updateUseOcclusionCompensation();
+
+  // Scan for available transport plugins
+  scanForTransportSubscriberPlugins();
+
+  pointcloud_common_->initialize(context_, scene_node_);
+  pointcloud_common_->xyz_transformer_property_->hide();
+
+  depth_topic_property_->initialize(context_->getRosNodeAbstraction());
+  color_topic_property_->initialize(context_->getRosNodeAbstraction());
+}
+
+DepthCloudDisplay::~DepthCloudDisplay()
+{
+  if (initialized()) {
+    unsubscribe();
+    delete pointcloud_common_;
+  }
+
+  if (ml_depth_data_) {
+    delete ml_depth_data_;
+  }
+}
+
+void DepthCloudDisplay::setTopic(const QString & topic, const QString & datatype)
+{
+  // Copied from ImageDisplayBase::setTopic()
+  if (datatype == "sensor_msgs::msgs::Image") {
+    depth_transport_property_->setStdString("raw");
+    depth_topic_property_->setString(topic);
+  } else {
+    int index = topic.lastIndexOf("/");
+    if (index == -1) {
+      return;
```

Comment:
This comment is still open.

Reply:
This method is filling some of the elements in the GUI, I don't think we should return something different, We are looking specific topics and this might generate unnecessary noise in the CMD.